### ISSUE TITLE
feat(rtrrl): expose standard SGD for every parameter block and entry

### ISSUE THREAD

**Issue by le2333** [le2333/RTRRL#72]
## 背景

Memo 的更新组件已经定义了标准 `Sgd` 和对应的 Optax transform，但 RTRRL 将优化器 family 限制为：

```python
RTRRL_OPTIMIZERS = STEP_FAMILY.restricted("adam", "d_rtrrl", "iu")
```

因此当前 RTRRL-LRU、RTRRL-RTU 以及复用该 family 的 RFLO 变体无法选择普通 SGD。SGD 应作为与 Adam、D-RTRRL、IU 并列的标准更新规则接入，而不是通过 D-RTRRL 参数模拟。

### 配置与组件

- 将底层现有 `sgd` 分支加入 RTRRL optimizer family。
- torso、actor、critic 三个参数块均可独立选择：

```yaml
optimizer:
  kind: sgd
  sgd:
    lr: ...
```

- 每个块拥有独立的 SGD learning rate 和 optimizer state；不能让 torso 的设置隐式控制两个 head。
- SGD 继续使用统一 `UpdateRule` contract，接收算法提供的 trace、direct direction 和 TD signal；不要在 RTRRL 内重新实现一套 SGD。
- 标准 SGD 不维护动量；本 issue 不新增 momentum/Nesterov 语义。若未来需要，应作为显式、可辨识的独立配置扩展。

### RTRRL 更新语义

- SGD 使用与当前 Adam 相同的 RTRRL 输入方向：算法先按现有规则组合 `delta * trace + direct`，对环境轴求均值，再由 SGD 乘 learning rate。
- 保留现有 ascent 方向约定，最终仍由算法执行 `parameter + update`；不得因直接调用 Optax SGD 而反转符号。
- torso 的 `grad_clip` 对 SGD 应与 Adam 一样作用于完成组合后的 torso update direction。
- actor、critic 当前没有额外 outer clip，接入 SGD 不应暗中增加。
- SGD 不改变资格迹的 reading、emphasis、reset、entropy direct term 或 followed torso copy 语义。
- 该接入与 #69 的 input/output IU/ObGD 聚合设计正交；不要借此改变 IU、D-RTRRL 或 ObGD 的配置含义。

### 入口覆盖

以下所有当前可用入口均应暴露 SGD，而不是只修改一个参数声明：

- RTRRL-LRU；
- RTRRL-RTU；
- RTRRL-CTRNN-RFLO；
- RTRRL-SSM-RFLO / RTRRL-LSTM-RFLO（如果其入口复用 RTRRL optimizer family）；
- 对应 single-run 和 ensemble/HPO 入口；
- deployment catalog 与配置 schema。

若某个 RFLO 入口使用独立 optimizer family，应显式接线并保持同一配置名称，不能静默缺失。

## 验收标准

- [ ] `torso.optimizer.kind: sgd`、`actor.optimizer.kind: sgd`、`critic.optimizer.kind: sgd` 可分别构建和运行。
- [ ] 单步数值测试：给定相同 trace、direct、delta 和 learning rate，RTRRL SGD update 与手算的 `lr * mean(delta * trace + direct, env)` 一致。
- [ ] 验证 ascent 符号正确，参数应用后目标方向与当前 Adam/SGD UpdateRule 约定一致。
- [ ] torso `grad_clip: 0` 与非零值分别和对应 Optax chain reference 一致。
- [ ] 混合配置可用，例如 torso=SGD、actor=Adam、critic=IU，各块状态和更新互不串线。
- [ ] `num_envs > 1` 下只在完成每个 stream 的方向后按现有语义求均值。
- [ ] episode reset、trace carry、checkpoint/resume 下 SGD 不改变算法状态语义。
- [ ] LRU、RTU 和所有已注册 RFLO 变体的 single/ensemble 构建测试覆盖 `sgd`。
- [ ] catalog 中相关 entry 的 optimizer 参数包含 `sgd.lr`，HPO space 能选择或固定 SGD。
- [ ] 旧 Adam、D-RTRRL、IU 配置和逐步数值结果不变。

## 非目标

- 不新增 momentum、Nesterov、weight decay 或学习率 schedule。
- 不把 ObGD 当作 SGD 的别名；ObGD 是带内部动态 bound 的独立规则。
- 不修改 #69 所定义的 input/output 聚合拓扑。
